Modifier/CategoryModifier.php

```diff
@@ -107,7 +107,7 @@ public function addVarcharAttributes($entity, CategoryDocument $document)
         $varcharAttributes = $entity->getVarcharAttributes();
 
         foreach ($varcharAttributes as $attribute) {
-            if ($this->storeId !== $attribute->getStore()) {
+            if ($this->storeId !== $attribute->getStore() && $attribute->getStore() !== 0) {
```

Comment:
maybe you want to add comment for this functionality?
as far as I understant its to skip item if its from different shop, but comment would make it more clear. Thats my opinion only :)